src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -324,11 +324,11 @@ public static ObjectFactory<T>
 #endif
             CreateFactoryInternal(typeof(T), argumentTypes, out ParameterExpression provider, out ParameterExpression argumentArray, out Expression factoryExpressionBody);
 
-            var factoryLambda = Expression.Lambda<Func<IServiceProvider, object?[]?, T>>(
+            var factoryLambda = Expression.Lambda<ObjectFactory<T>>(
```

Comment:
Note for other reviewers: `ObjectFactory<T>` delegate is declared here

https://github.com/dotnet/runtime/blob/da757a1b062c24e1011c5aadf31fe2f1cde59c97/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ObjectFactoryT.cs